### PR TITLE
fix: The colon disappears if time field is updating manually (M2-6008)

### PR DESCRIPTION
**Pull request by nmialik**
- [x] Tests for the changes have been added
- [ ] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-6008](https://mindlogger.atlassian.net/browse/M2-6008)

### CODE REVIEW

File: src/modules/Dashboard/features/Applet/Schedule/EventForm/AvailabilityTab/AvailabilityTab.tsx [hunk 1]
```diff
@@ -195,6 +195,7 @@ export const AvailabilityTab = ({
                 label={t('from')}
                 onCustomChange={(time) => handleTimeCustomChange(time, TimeType.FromTime)}
                 data-testid={`${dataTestid}-start-time`}
+                defaultTime="00:00"
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Applet/Schedule/EventForm/AvailabilityTab/AvailabilityTab.tsx]: Please use constants from `EventForm.const.tsx`. In other occurrences as well.

image

File: src/shared/components/TimePicker/TimePicker.tsx [hunk 2]
```diff
@@ -21,58 +24,99 @@ export const TimePicker = <T extends FieldValues>({
...
+  const handleKeyDownEnter = (
+    event: KeyboardEvent<HTMLDivElement>,
+    onChange: (value: string) => void,
+  ) => {
+    if (event.key !== 'Enter') return;
+    handleInputValue(onChange);
+  };
+
+  const handleChange = (
+    date: Date | null,
```
Comment (IhorNesterenko99) [inline src/shared/components/TimePicker/TimePicker.tsx]: It's better to use object for types if there are more than two arguments. 
And please move types to `TimePicker.types.ts`. 
The type for the `onChange` (`(value: string) => void`) can be reused.

File: src/shared/components/TimePicker/TimePicker.tsx [hunk 3]
```diff
@@ -21,58 +24,99 @@ export const TimePicker = <T extends FieldValues>({
...
+                placeholderText={placeholder}
+                autoComplete="off"
+                customInput={
+                  <TextField
+                    variant="outlined"
+                    label={label}
+                    error={!!error}
+                    helperText={error?.message || null}
+                    inputProps={{
+                      maxLength: 5,
```
Comment (IhorNesterenko99) [inline src/shared/components/TimePicker/TimePicker.tsx]: Please move to the const in the `TimePicker.const.ts`.

File: src/shared/components/TimePicker/TimePicker.tsx [hunk 4]
```diff
@@ -21,58 +24,99 @@ export const TimePicker = <T extends FieldValues>({
...
+  const updateInputValue = (value: string, onChange: (value: string) => void) => {
+    onCustomChange?.(value);
+    setInputValue(value);
+    onChange(value);
+  };
+
+  const handleInputValue = (onChange: (value: string) => void) => {
+    const cleanedInput = cleanInput(inputValue);
+    const validatedInput = validateInput(defaultTime, cleanedInput);
+    const formattedInput = validatedInput.length === 4 ? formatInput(validatedInput) : defaultTime;
```
Comment (IhorNesterenko99) [inline src/shared/components/TimePicker/TimePicker.tsx]: If I'm typing '12' and press enter, is it expected to output '12:00'? Maybe it's also good idea to take such cases into account. 

image

File: src/shared/components/TimePicker/TimePicker.tsx [hunk 5]
```diff
@@ -8,9 +8,12 @@ import { DateFormats } from 'shared/consts';
 
 import { StyledIcon, StyledTimePickerWrapper } from './TimePicker.styles';
 import { TimePickerProps } from './TimePicker.types';
+import { cleanInput, formatInput, validateInput } from './TimePicker.utils';
 
 const ReactDatePicker = lazy(() => import('react-datepicker'));
 
+const DEFAULT_TIME = '23:59';
```
Comment (IhorNesterenko99) [inline src/shared/components/TimePicker/TimePicker.tsx]: Please move to `TimePicker.const.ts`.

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**